### DISCUSSION TITLE
pnp webpart v4 4.7 Search box do not show any results

### DISCUSSION THREAD

**Discussion by MayurHalarnkar** [microsoft-search/pnp-modern-search#2497] [General]
pnp webpart v4 4.7 Search box do not show any results, guys any idea > I am using June 2022 release. When we connect search box with search results it gives no results, but without search box beign connected shows full records.

**Comment by MayurHalarnkar** [comment]
Pnp Search webpart Searchbox issue

1.	We are using pnp Search V4 webpart in our environment
 
 
2.	We are able to configure Search Results properly as below. We can see that we are able to fetch 3441 records using 
 
 
  





3.	I am facing issue when I connect Search results with pnp Searchbox with below settings. Not able to search a single record even if keyword do exists in search results.

 

Result output is null

 

 

failed to process execution entry:Error: Cannot perform DeepCopy() because a circular reference was encountered, object: [object Object], property: helpers
e._processExecutionEntry @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.addExecutionEntry @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.writeQosEndResult @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e._writeQosEnd @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.writeSuccess @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e._getPropertyValue @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e._getData @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e._getData @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.tryGetValue @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
tryGetValueSafe @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
(anonymous) @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
mergeFiltersDataFromSources @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
renderCompleted @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
(anonymous) @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
o @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
Promise.then (async)
c @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
(anonymous) @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
R @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
render @ modern-search-filters-web-part_80384cefa4236ce1169d.js:2
(anonymous) @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e._raiseEventInternal @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.raiseEvent @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.notifyPropertyChanged @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.notifyPropertyChanged @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
_onDataRetrieved @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
(anonymous) @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
o @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
Promise.then (async)
c @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
o @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
Promise.then (async)
c @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
(anonymous) @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
M @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
getDataFromDataSource @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
(anonymous) @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
(anonymous) @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
M @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
componentDidUpdate @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
ss @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
bc @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
t.unstable_runWithPriority @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Gi @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
hc @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
ac @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
(anonymous) @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
t.unstable_runWithPriority @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Gi @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Qi @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
qi @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Zs @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Bc @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Wc @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
t.render @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
renderCompleted @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
(anonymous) @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
o @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
Promise.then (async)
c @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
o @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
Promise.then (async)
c @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
o @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
Promise.then (async)
c @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
(anonymous) @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
Er @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
render @ modern-search-results-web-part_9402d0e6f3b644596f93.js:2
t._dynamicPropertyRefresh @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
(anonymous) @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
(anonymous) @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e._raiseEventInternal @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.raiseEvent @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.notifyPropertyChanged @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
e.notifyPropertyChanged @ sp-pages-assembly_en-us_96f14ace9443d51fd58e58cb1f2b4b36.js:70
K._onSearch @ modern-search-box-web-part_8ed1f231ce381def07ce.js:2
i @ modern-search-box-web-part_8ed1f231ce381def07ce.js:2
(anonymous) @ modern-search-box-web-part_8ed1f231ce381def07ce.js:2
_onSearch @ modern-search-box-web-part_8ed1f231ce381def07ce.js:2
onSearch @ modern-search-box-web-part_8ed1f231ce381def07ce.js:2
n._onKeyDown @ modern-search-box-web-part_8ed1f231ce381def07ce.js:2
s @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
p @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
(anonymous) @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
b @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
rt @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
at @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
ot @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
ft @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
H @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
j @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Xt @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Jt @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
t.unstable_runWithPriority @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Gi @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
U @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2
Qt @ react-dom-16-bundle_none_8eaea77e9a62cc734837.js:2

4.	Filters are working perfectly , can you please help us what might be the issue with search box?

**Reply by wobba** [reply]
Not sure what the issue is.. can you test with 4.8-beta? And did you connct both ways? https://microsoft-search.github.io/pnp-modern-search/scenarios/create-simple-search-page/

**Comment by MayurHalarnkar** [comment]
[PNP Searchbox data fetch issue with Pnp Search webpart v4.docx](https://github.com/microsoft-search/pnp-modern-search/files/9893405/PNP.Searchbox.data.fetch.issue.with.Pnp.Search.webpart.v4.docx)

**Comment by nthoman** [comment]
I had the same issue with 4.7.
https://github.com/microsoft-search/pnp-modern-search/discussions/2409

**Reply by nthoman** [reply]
@wobba the 4.8 beta seems to have fixed this issue.

**Comment by MayurHalarnkar** [comment]
PROTECTED

Hi All,

This issue has been resolved .

It was a space required after {SearchTerms} in query template.

Regards

Mayuresh Halarnkar
